### ISSUE TITLE
Page reload issues

### ISSUE THREAD

**Issue by andrewfam** [yourlabs/django-session-security#39]
Hi,

I have some pages with a reload page upon ajax success. 

{% include 'session_security/all.html' %}
Adding in the include into the page throws up the alert.

You have unsaved changes in a form of this page.
Are you sure you want to reload this page?
![screen shot 2015-02-05 at 2 52 17 pm](https://cloud.githubusercontent.com/assets/1430616/6056068/a342f2ac-ad46-11e4-82c3-9f7e0de0ba80.png)

Would you know how to fix this?

**Comment by jpic** [comment]
Override that template to pass `confirmFormDiscard = false`.

https://github.com/yourlabs/django-session-security/blob/stable/2.x.x/session_security/templates/session_security/all.html
https://github.com/yourlabs/django-session-security/blob/stable/2.x.x/session_security/static/session_security/script.js#L38

**Comment by anthonybu** [comment]
I have the same issue:
The form is only considered to be "clean" when submitted. I enhanced the logic to include reset case.
So after your Ajax calls, just call .reset() on the form. It should clear out the unsaved change message.

https://github.com/yourlabs/django-session-security/pull/42

**Comment by anthonybu** [comment]
Hi James,

Since this is fixed, would you push an update out so we can pip update?

Thanks!

**Comment by psychok7** [comment]
will this fix this issue as well?

![screen](https://cloud.githubusercontent.com/assets/282284/8482068/070df938-20df-11e5-9c75-b0412c55b9c7.png)

**Comment by anthonybu** [comment]
@psychok7 , first of all, I don't know if the code fix has made to the pip repo yet, could you double check if this code change is included in your local library:
https://github.com/anthonybu/django-session-security/commit/88550eff7e5edfe439abcffe0ec6f066ebe068b7

if the code is there, you just need to call form.reset() before page reload/redirect.

**Comment by psychok7** [comment]
@anthonybu according to this its there already https://github.com/yourlabs/django-session-security/compare/2.2.3...stable/2.x.x

i was thinking of writing a setting to disable that,  something like `SESSION_SECURITY_CONFIRM_FORM_DISCARD` do you know if there are other implications if i do that?

**Comment by anthonybu** [comment]
@psychok7 I don't think so, but you better check with @jpic .

Personally, I like the little warning. But giving the user a way to disable it globally is not a bad idea.

**Comment by johnfraney** [comment]
I was able to achieve this behaviour without overriding any templates by setting `sessionSecurity.confirmFormDiscard` to `undefined` rather than `false`:

```
  {% include 'session_security/all.html' %}
  <script>
    sessionSecurity.confirmFormDiscard = undefined;
  </script>
```

I could do a PR to update the docs if that would be helpful.

**Comment by jpic** [comment]
Thanks @johnfraney that would be very welcome.

I don't think that doing a django setting to configure a javascript variable is a good idea, it'll be more code to test while you can just configure the js var in js itself.

As for having some sort of user-settings page, I'm all for it, but please open a separate issue for that.

**Comment by psychok7** [comment]
@johnfraney the way your doing is in fact the easiest way

**Comment by jpic** [comment]
Sorry guys, I should have documented it better, it seemed so natural to
me​...

**Comment by johnfraney** [comment]
@psychok7 I like easy!
@jpic No problem at all. When someone writes a great plugin, the least I can do to give back is add a bit of documentation.

**Comment by jpic** [comment]
Fixed by #54
Thanks !
